apps/contracts/contracts/ZKGroups.sol

```diff
@@ -3,108 +3,86 @@ pragma solidity ^0.8.4;
-/// zk-groups dashboard allow admins to create on-chain groups by directly connect to official Semaphore contract.
-/// A Zk-groups contract save the Merkle tree roots of the other off-chain groups(i.e. reputation and permissioned groups),
-/// which will then allow members of off-chain groups to verify their zero-knowledge proofs in that contract.
-contract ZKGroups is IZKGroups, SemaphoreCore, Ownable {
-    /// @dev Gets a merkle tree depth and returns its verifier address.
-    mapping(uint256 => IVerifier) public verifiers;
-
-    /// @dev Gets a group id and returns the off-chain group parameters.
-    mapping(bytes32 => OffchainGroup) public offchainGroups;
-
-    /// @dev Gets a group id and returns boolean about nullifierHashes used.
-    mapping(bytes32 => mapping(uint256 => bool)) internal nullifierHashes;
+import "@semaphore-protocol/contracts/interfaces/ISemaphoreVerifier.sol";
 
-    /// @dev Checks if there is a verifier for the given merkle tree depth.
-    /// @param merkleTreeDepth: Depth of the merkle tree.
-    modifier onlySupportedMerkleTreeDepth(uint256 merkleTreeDepth) {
-        if (address(verifiers[merkleTreeDepth]) == address(0)) {
-            revert("MerkleTreeDepth is not supported");
-        }
-        _;
-    }
+/// @title ZKGroups
+/// @dev This contract is used to save the Merkle roots of the off-chain groups.
+contract ZKGroups is IZKGroups, Ownable {
+    ISemaphoreVerifier public verifier;
 
-    /// @dev Initializes the Semaphore verifiers used to verify the user's ZK proofs.
-    /// @param _verifiers: List of Semaphore verifiers (address and related Merkle tree depth).
-    constructor(Verifier[] memory _verifiers) {
-        for (uint8 i = 0; i < _verifiers.length; ) {
-            verifiers[_verifiers[i].merkleTreeDepth] = IVerifier(_verifiers[i].contractAddress);
+    /// @dev Gets a group name and returns the off-chain group parameters.
+    mapping(bytes32 => Group) public groups;
 
-            unchecked {
-                ++i;
-            }
-        }
+    /// @dev Gets a group name and a nullifier hash and returns true if it has already been used.
+    mapping(bytes32 => mapping(uint256 => bool)) internal nullifierHashes;
+
+    /// @dev Initializes the Semaphore verifier used to verify the user's ZK proofs.
+    /// @param _verifier: Semaphore verifier address.
+    constructor(ISemaphoreVerifier _verifier) {
+        verifier = _verifier;
     }
 
-    /// @dev See {IZKGroups-updateOffchainGroups}.
-    function updateOffchainGroups(OffchainGroup[] calldata _offchainGroups) external override onlyOwner {
-        for (uint8 i = 0; i < _offchainGroups.length; ){
-            _updateOffchainGroup(_offchainGroups[i].groupName, _offchainGroups[i]);
+    /// @dev See {IZKGroups-updateGroups}.
+    function updateGroups(bytes32[] calldata groupNames, Group[] calldata _groups) external override onlyOwner {
+        for (uint256 i = 0; i < groupNames.length; ){
```

Comment:
How about adding a constraint to verify `groupNames.length` is same as `_groups.length` 🤔

Reply:
Yes absolutely! Thanks